### PR TITLE
Bump version to v1.2.4-rc2

### PR DESCRIPTION
**Pull request by nishanttotla**
cc @dongluochen @abronan 

Signed-off-by: Nishant Totla <nishanttotla@gmail.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,43 @@
...
 
-## 1.2.4 (2016-06-14)
+## 1.2.4 (2016-07-20)
 
 #### API
 
 - Code refactoring
+- Additional error handling
+- Update minimum Docker Engine version supported by Swarm to 1.8
+- New client interface in Swarm, to differentiate from Swarm mode in Docker 1.12
```
Comment (dongluochen) [inline CHANGELOG.md]: Maybe better to reorder them by significance. The last item has bigger impact.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,43 @@
...
+
+#### Networking
+
+- Fix concurrent map writes race condition
+- Refresh single network when network event emitted (performance improvement)
```
Comment (dongluochen) [inline CHANGELOG.md]: "emitted" -> "is emitted" or "emits"

**Comment by dongluochen** [comment]
LGTM

**Comment by nishanttotla** [comment]
@dongluochen addressed your comments.

**Comment by vieux** [comment]
LGTM

**Pull request merged**